Core/Src/main.c

```diff
@@ -75,6 +77,10 @@ int main(void)
 
   /* USER CODE BEGIN Init */
 
+  uint8_t T_buffer [3] = {0x01, 0x80, 0};
+  uint8_t R_buffer [3] = {0};
+  uint16_t adc_value [1] = {0};
```

Comment:
You don't always have to have variables as arrays! For `adc_value` since you only plan on storing and modifying one value, this can just be a regular variable and not array.